classes/statistics/MetricsDAO.inc.php

```diff
@@ -94,9 +94,241 @@ protected function getAssocObjectInfo($submissionId, $contextId) {
 		$publishedArticleDao = DAORegistry::getDAO('PublishedArticleDAO');
 		$publishedArticle = $publishedArticleDao->getByArticleId($submissionId, $contextId, true);
 		if ($publishedArticle) {
-			$returnArray = array(ASSOC_TYPE_ISSUE, $publishedArticle->getIssueId());
+			$returnArray = [ASSOC_TYPE_ISSUE, $publishedArticle->getIssueId()];
 		}
 		return $returnArray;
 	}
-}
 
+	/**
+	 * Retrieves several statistics about submissions, with sums and averages
+	 * @param $contextId int The context id
+	 * @param $dateStart string The start date filter
+	 * @param $dateEnd string The end date filter
+	 * @param $sectionIds array A list of section IDs to filter the result set
+	 * @return array Returns a dictionary with statistics
+	 */
+	public function getSubmissionStatistics($contextId = null, $dateStart = null, $dateEnd = null, $sectionIds = null) {
+		// Access submission constants.
+		import('lib.pkp.classes.submission.Submission');
+
+		// Access decision actions constants.
+		import('classes.workflow.EditorDecisionActionsManager');
+
+		$params = $where = $whereMinimum = [];
+		if($dateStart || $dateEnd) {
+			$where[] = 'ed.date_decided ' . ($dateStart && $dateEnd ? 'BETWEEN ? AND ?' : ($dateStart ? '>= ?' : '<= ?'));
+			$dateStart && $params[] = $dateStart;
+			$dateEnd && $params[] = $dateEnd;
+		}
+
+		if ($contextId) {
+			$where[] = 's.context_id = ?';
+			$params[] = $contextId;
+		}
+
+		if (count($sectionIds)) {
+			$where[] = 's.section_id IN (' . substr(str_repeat(',?', count($sectionIds)), 1) . ')';
+			$params = array_merge($params, $sectionIds);
+		}
+
+		$params = array_merge($params, [$dateEnd, $dateStart]);
+
+		$sql = '
```

Comment:
The SQL clauses need a good review, I lost edit access to the specification (so I can't see the comments) and I remember of meeting some strange conditions in the database (e.g. published submission with declined as last decision), so I've decided to let some conditions less strict.